### PR TITLE
admanmedia ad network type added

### PR DESCRIPTION
**Pull request by luciancrasovan**
AdMan Media ad network type implementation

- This implements basic amp-ad for admanmedia with a single data parameter
- Unique size ads: 300 X 250
- No layout configuration
- No viewability control

### CODE REVIEW

**Comment by luciancrasovan** [comment]
I signed it!

**Comment by luciancrasovan** [comment]
It seems CI build failed but I cannot see a relation with my changes. Please confirm and let me know which would be the solution. Thanks in advance!

**Comment by luciancrasovan** [comment]
This is for the AdMan Media company [http://admanmedia.com/](url)
It urges us to see it available in the Repository in order to be able to test it with real clients so a fast acceptance of this pull request will be highly appreciated!

**Comment by lcrasovan** [comment]
Hello,

Is any news about this pull request?

Thanks in advance

**Comment by jasti** [comment]
@lannka @zhouyx would you please review this PR? Thanks.

File: examples/ads.amp.html [hunk 1]
```diff
@@ -247,6 +248,12 @@ <h2 id="adman">Adman</h2>
       data-host="talos.adman.gr">
   </amp-ad>
 
+  <h2 id="admanmedia">AdmanMedia</h2>
+  <amp-ad width="300" height="250"
+          type="admanmedia"
```
Comment (zhouyx) [inline examples/ads.amp.html]: I'm not able to see a rendered example. Could you please check that the example ad is rendering?

**Comment by lcrasovan** [comment]
Hello @zhouyx , we have just uploaded our changes. Could you please review and let us know if everything is ok?

File: ads/admanmedia.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# Admanmedia 
+
+Please visit our web (http://www.admanmedia.com) for more details
+## Example
+
+```html
+<amp-ad width="300" height="250"
+    type="admanmedia"
+    data-id="8e916419">
+  </amp-ad>
```
Comment (zhouyx) [inline ads/admanmedia.md]: nit: indentation here.

File: ads/admanmedia.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import {validateData, loadScript} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function admanmedia(global, data) {
+  validateData(data, ['id']);
+
+  loadScript(global, `https://mona.admanmedia.com/go?id=${data.id}`, () => {
```
Comment (zhouyx) [inline ads/admanmedia.js]: please use `encodeURIComponent(data.id)`

File: examples/ads.amp.html [hunk 4]
```diff
@@ -265,6 +266,12 @@
       data-host="talos.adman.gr">
   </amp-ad>
 
+  <h2>AdmanMedia</h2>
+  <amp-ad width="300" height="250"
```
Comment (zhouyx) [inline examples/ads.amp.html]: I get error when testing locally. Could you please take a look? 
image

Comment (lcrasovan) [inline examples/ads.amp.html]: When testing locally, as the mona.admanmedia.com does not set specially CORS headers for localhost, I am launching Chrome like this: ` ./chrome --disable-web-security --user-data-dir `
Prior to launching it to production we will setup CORS headers on our server and this error will not be launched anymore

File: ads/admanmedia.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function admanmedia(global, data) {
+  validateData(data, ['id']);
+
+  const encodedId = encodeURIComponent(data.id);
+  loadScript(global, `https://mona.admanmedia.com/go?id=${encodedId}`, () => {
+    const pattern = `script[src$="id=${data.id}"]`;
```
Comment (zhouyx) [inline ads/admanmedia.js]: encodedId here as well?

File: examples/ads.amp.html (hunk 4 above)
Comment (zhouyx) [inline examples/ads.amp.html]: Cool I am able to see it now. After this Pull Request is merged it usually takes two weeks to hit prod. It's usually towards the end of a week, but not always on an exact data. Is there a concern why you can't setup the header a bit earlier?

**Comment by zhouyx** [comment]
Look good in general. Few minor requests.

File: examples/ads.amp.html (hunk 4 above)
Comment (lcrasovan) [inline examples/ads.amp.html]: Thanks for you valuable remarks that helped us improve the code quality. We have started the internal process for CORS header setting and hopefully in a few days we will be all set.

**Comment by zhouyx** [comment]
LGTM.  Thanks for the Pull Request. Merged.

**Pull request merged**